ethereum/pow/ethpow.py

```diff
@@ -139,7 +140,7 @@ def mine(
 
     cache = cache_gen(cache_size, block_number)
     nonce = start_nonce
-    target = zpad(int_to_big_endian(2 ** 256 // (difficulty or 1) - 1), 32)
+    target = (2 ** 256 // (difficulty or 1)).to_bytes(32, byteorder="big")
```

Comment:
this is the difference, previously it won't guarantee 32 bits. another fix is keep `target` as int, and parse `o[b'result']` at line 148 to int every time for comparison.

also added a test to prevent previous error

Reply:
seems ethash utils assume little endian